lib/execution_test.go

```diff
@@ -48,4 +54,104 @@ func TestExecutionStateVUIDs(t *testing.T) {
 	assert.Equal(t, uint64(4+count), es.GetUniqueVUIdentifier())
 }
 
-//TODO: way more tests...
+func TestExecutionStateGettingVUsWhenNonAreAvailable(t *testing.T) {
+	t.Parallel()
+	es := NewExecutionState(Options{}, 0, 0) //TODO: fix
+	logHook := &testutils.SimpleLogrusHook{HookedLevels: []logrus.Level{logrus.WarnLevel}}
+	testLog := logrus.New()
+	testLog.AddHook(logHook)
+	testLog.SetOutput(ioutil.Discard)
+	vu, err := es.GetPlannedVU(logrus.NewEntry(testLog), true)
+	require.Nil(t, vu)
+	require.Error(t, err)
+	require.Contains(t, err.Error(), "could not get a VU from the buffer in")
+	entries := logHook.Drain()
+	require.Equal(t, MaxRetriesGetPlannedVU, len(entries))
+	for _, entry := range entries {
+		require.Contains(t, entry.Message, "Could not get a VU from the buffer for ")
```

Comment:
1. This is testing 2 different things:
1.1. the error is if we decide that we just can't continue to run the test because we can't get a VU
1.2. the log is when for a certain period of time we weren't able to get a VU (but we should've been able) so that it's know that ... the test might not be very okay ... .Maybe this should also emit a metric @na-- ?
2. I don't see why if someone decides to change the code they shouldn't also have to change the test for that code given that ... this is literally in the code I am testing not in some other code that gets called by it

Reply:
@MStoykov maybe, though I think this would either be covered by the future CPU/RAM usage metrics, or it would be a bug in k6 (in which case k6 should abort with an error code).

Reply:
Yeah, a metric or an error would make more sense if this should be a separate functional scenario (1.2 in @MStoykov's comment). Relying on logging for testing functionality is a bit smelly to me. :)

> I don't see why if someone decides to change the code they shouldn't also have to change the test for that code given that ... this is literally in the code I am testing not in some other code that gets called by it

True, it's not a big deal in this case since the logging happens directly in the tested function, but as suggested, maybe this should additionally emit a metric or return an error.

I'll resolve this as it can be addressed later.